### PR TITLE
Remove ActiveSupport and DS_Store

### PR DESCRIPTION
**Pull request by colbyh**
- removes FactoryGirl in favor of Fabricate so we can move away from ActiveSupport
- removes all hash methods that depend on ActiveSupport
- removes all object creation methods that depend on ActiveSupport
- adds XML matcher to verify XML using only Faraday (instead of ActiveSupport)
- creates fabrications for all major models
- updates naming conventions to be more predictable
- updates models to use `attr_acessor` instead of reader for greater flexibility

[#122086039]

### CODE REVIEW

**Comment by colbyh** [comment]
tag @rheaton

File: ingram_micro.gemspec [hunk 1]
```diff
@@ -6,8 +6,8 @@ require 'ingram_micro/version'
 Gem::Specification.new do |spec|
   spec.name          = "ingram_micro"
   spec.version       = IngramMicro::VERSION
-  spec.authors       = ["Rachel Heaton","Joseph Nguyen", "Ben Christel"]
-  spec.email         = ["rachelmheaton@gmail.com"]
+  spec.authors       = ["Ben Christel", "Rachel Heaton", "Colby Holliday", "Joseph Nguyen"]
+  spec.email         = ["webops@walkerandcobrands.com"]
```
Comment (rheaton) [inline ingram_micro.gemspec]: gracias.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-      config.customer_id = '123'
+      config.api_root = 'http://example.com'
+      config.partner_name = Faker::Company.name
+      config.partner_password = Faker::Internet.password
+      config.customer_id = (1..100).to_a.sample.to_s
       config.debug = false
       config.proxy = nil
     end
-
-    IngramMicro.configuration.assert_valid
```
Comment (rheaton) [inline spec/spec_helper.rb]: Y no assert valid?

File: spec/support/matchers.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+RSpec::Matchers.define :have_xml do |xpath, text|
```
Comment (rheaton) [inline spec/support/matchers.rb]: :yey:

**Comment by rheaton** [comment]
Looks good Troll! @colbyh

**Pull request merged**